src/libtorch.cc

```diff
@@ -587,8 +590,49 @@ ModelInstanceState::ValidateTypedSequenceControl(
 }
 
 TRITONSERVER_Error*
-ModelInstanceState::ValidateInputs()
+ModelInstanceState::ValidateInputs(const size_t expected_input_cnt)
 {
+  const torch::jit::Method& method = torch_model_->get_method("forward");
+  const auto& schema = method.function().getSchema();
+  const std::vector<c10::Argument>& arguments = schema.arguments();
+
+  // Currently we only support input of type Dict[str, Tensor] when the
+  // model expects a single input. If the model expects more than one input then
+  // they must be all be of type Tensor.
+  //
+  // Ignore the argument at idx 0 if it is of Class type (self param in forward
+  // function)
+  size_t start_idx = 0;
+  if ((arguments.size() > 0) &&
+      (arguments.at(0).type()->kind() == c10::TypeKind::ClassType)) {
+    start_idx = 1;
+  }
+  if ((arguments.size() == (1 + start_idx)) &&
+      (arguments.at(start_idx).type()->kind() == c10::TypeKind::DictType)) {
+    is_dict_input_ = true;
+  } else if (arguments.size() > start_idx) {
+    for (size_t i = start_idx; i < arguments.size(); i++) {
+      if (arguments.at(i).type()->kind() != c10::TypeKind::TensorType) {
+        return TRITONSERVER_ErrorNew(
+            TRITONSERVER_ERROR_INTERNAL,
+            "The model must expect a single input of type Dict[str, Tensor] "
```

Comment:
Error can be improved. "Multiple non-c10::TypeKind::TensorType inputs detected in the model. Only single  c10::TypeKind::DictType input is supported besides c10::TypeKind::TensorType inputs".  Or somehting like this

Reply:
I want to use Dict[str, Tensor] since the Dict must have string and Tensor as key and value. The reason for not using c10::TypeKind is that I want this to match the convention used by PyTorch when logging the graph.

```
method forward {
  input self    ,__torch__.AddSubNet
  input input0.1        ,Dict(str, Tensor)
  }
 ```